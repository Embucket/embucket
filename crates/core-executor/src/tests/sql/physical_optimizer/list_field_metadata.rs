use crate::test_query;

test_query!(
    list_field_metadata_array,
    "SELECT * FROM array",
    setup_queries = ["CREATE TABLE array as (SELECT [1,2]::ARRAY)"],
    snapshot_path = "list_field_metadata"
);

test_query!(
    list_field_metadata_literal,
    "SELECT * FROM array",
    setup_queries = ["CREATE TABLE array as (SELECT 'A'::ARRAY)"],
    snapshot_path = "list_field_metadata"
);

test_query!(
    list_field_metadata_arrays_column,
    "SELECT * FROM array",
    setup_queries =
        ["CREATE TABLE array as (SELECT arr FROM VALUES ([1,2]::ARRAY),([1,2]::ARRAY) AS t(arr))"],
    snapshot_path = "list_field_metadata"
);

test_query!(
    list_field_metadata_arrays_with_column_cast,
    "SELECT * FROM array",
    setup_queries =
        ["CREATE TABLE array as (SELECT arr::ARRAY as arr FROM VALUES ([1,2]),([1,2]) AS t(arr))"],
    snapshot_path = "list_field_metadata"
);
