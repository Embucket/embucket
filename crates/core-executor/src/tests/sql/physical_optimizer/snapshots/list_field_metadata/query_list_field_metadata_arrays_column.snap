---
source: crates/core-executor/src/tests/sql/physical_optimizer/list_field_metadata.rs
description: "\"SELECT * FROM array\""
info: "Setup queries: CREATE TABLE array as (SELECT arr FROM VALUES ([1,2]::ARRAY),([1,2]::ARRAY) AS t(arr))"
---
Ok(
    [
        "+--------+",
        "| arr    |",
        "+--------+",
        "| [1, 2] |",
        "| [1, 2] |",
        "+--------+",
    ],
)
